### ISSUE TITLE
[CT-2626] [Regression] dbt snapshot does not fetch the target path from environment variables

### ISSUE THREAD

**Issue by ADraginda** [dbt-labs/dbt-core#7760]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

dbt 1.5.1

* we don't have target_path in our profiles.yml
* we have the env var: `'DBT_TARGET_PATH': '/tmp/198e1313-1c7f-44d7-92a5-9556f5d203bd'`
* but when we run `dbt --debug snapshot --model <model_name> --target production`  we see: 

```
running dbt with arguments {... 'target_path': 'None', ... }
```

### Expected Behavior

I would expect the target_path set in the environment variable respected (Note that LOG_FILE_PATH _is_ respected from the environment variables)

### Steps To Reproduce

```bash
export DBT_TARGET_PATH="/tmp/198e1313-1c7f-44d7-92a5-9556f5d203bd"
dbt --debug snapshot --model <model_name>
```

### Environment

```markdown
- OS: Centos (AWS MWAA 2.4.3)
- Python: 3.10.10
- dbt: 1.5.1
```

### Which database adapter are you using with dbt?

snowflake

**Comment by dbeatty10** [comment]
Thanks for reaching out @ADraginda !

I tried out each of the configuration options [here](https://docs.getdbt.com/reference/project-configs/target-path#configuration), and I observed something similar as you reported, but slightly different.

For the third option (`target-path` in `dbt_project.yml`), it didn't log out the target directory and printed `None` instead. As it seems you already know, this option is is deprecated in 1.5. Backward compatibility is still supported in 1.5 but will be removed in a future update.

But in all three cases, it appeared to create and use the target directory correctly (despite the `--debug` output for the deprecated option).

```shell
export DBT_TARGET_PATH=target_environment_variable                                                                                         
echo $DBT_TARGET_PATH
dbt --debug run -s my_model
```

Output:
```
target_environment_variable

21:31:53  Running with dbt=1.5.0
21:31:53  running dbt with arguments {... 'target_path': 'target_environment_variable', ...}
```

Is it not writing to the target directory specified in `DBT_TARGET_PATH` for you?

Even when `/tmp/198e1313-1c7f-44d7-92a5-9556f5d203bd` didn't already exist, it successfully wrote there on my machine.

**Comment by ADraginda** [comment]
Apologies - I have updated my ticket with the actual command I sent: It should have been `dbt snapshot` not `dbt run`. I just tested again and indeed, the snapshot was not using my environment variable `DBT_TARGET_PATH` (it reports as None, and does not create the folder, instead just creating `/target` in my current path)

**Comment by dbeatty10** [comment]
Ah, I see @ADraginda.

It looks like https://github.com/dbt-labs/dbt-core/pull/7419 was accidentally not yet included in 1.5.0 or 1.5.1, so we'll need to backport it within an upcoming release. Once we backport and release it, I believe it will resolve the issue you are reporting.

**Comment by ADraginda** [comment]
Wonderful, thanks so much @dbeatty10 !
